tests/golos.publication_rewards_tests.cpp

```diff
@@ -1194,9 +1201,61 @@ BOOST_FIXTURE_TEST_CASE(golos_delegators_test, reward_calcs_tester) try {
+    BOOST_CHECK_GT(vest.get_balance_raw(_users[8])["vesting"].as<asset>(), delegator1_amount);
+    BOOST_CHECK_GT(vest.get_balance_raw(_users[9])["vesting"].as<asset>(), delegator2_amount);
+    BOOST_CHECK_EQUAL(vest.get_balance_raw(_users[10])["vesting"].as<asset>(), delegator3_amount);
+    check();
+
+    BOOST_TEST_MESSAGE("--- add_funds_to_forum");
+    BOOST_CHECK_EQUAL(success(), add_funds_to_forum(50000));
+    check();
+    
+    BOOST_TEST_MESSAGE("--- checking that interest_rate < 1");
+    BOOST_TEST_MESSAGE("--- create_message: " << name{_users[0]}.to_string());
+    BOOST_CHECK_EQUAL(success(), create_message({_users[0], "permlink5"}));
+    check();
+    BOOST_CHECK_EQUAL(success(), delegate_vest(_users[11], _users[1], amount, interest_rate_1prcnt));
+    BOOST_CHECK_EQUAL(success(), delegate_vest(_users[12], _users[1], amount, interest_rate_1prcnt));
+    BOOST_CHECK_EQUAL(success(), delegate_vest(_users[13], _users[1], amount, interest_rate));
+    check();
+    BOOST_TEST_MESSAGE("--- " << name{_users[1]}.to_string() << " voted");
+    BOOST_CHECK_EQUAL(success(), addvote(_users[1], {_users[0], "permlink5"}, 10000));
+    check();
+    voter_amount = vest.get_balance_raw(_users[1])["vesting"].as<asset>();
+    delegator1_amount = vest.get_balance_raw(_users[11])["vesting"].as<asset>();
+    delegator2_amount = vest.get_balance_raw(_users[12])["vesting"].as<asset>();
+    delegator3_amount = vest.get_balance_raw(_users[13])["vesting"].as<asset>();
+    produce_blocks(golos::seconds_to_blocks(post.window));
+    BOOST_CHECK_GT(vest.get_balance_raw(_users[1])["vesting"].as<asset>(), voter_amount);
+    BOOST_CHECK_EQUAL(vest.get_balance_raw(_users[11])["vesting"].as<asset>(), delegator1_amount);
+    BOOST_CHECK_EQUAL(vest.get_balance_raw(_users[12])["vesting"].as<asset>(), delegator2_amount);
+    BOOST_CHECK_GT(vest.get_balance_raw(_users[13])["vesting"].as<asset>(), delegator3_amount);
+    check();
+
+    BOOST_TEST_MESSAGE("--- add_funds_to_forum");
+    BOOST_CHECK_EQUAL(success(), add_funds_to_forum(50000));
+    check();
+
+    BOOST_TEST_MESSAGE("--- increase vesting amount for " << name{_users[15]}.to_string());
+    add_vesting({_users[15]}, vest_amount.get_amount());
+    
+    BOOST_TEST_MESSAGE("--- checking that interest_rate doesn't overflow and curator has no reward");
+    BOOST_TEST_MESSAGE("--- create_message: " << name{_users[0]}.to_string());
+    BOOST_CHECK_EQUAL(success(), create_message({_users[0], "permlink6"}));
+    check();
+    BOOST_TEST_MESSAGE("--- retire vesting for curator");
+    BOOST_CHECK_EQUAL(success(), retire_vest(vest.get_balance_raw(_users[14])["vesting"].as<asset>(), _users[14], _issuer));
+    BOOST_CHECK_EQUAL(success(), delegate_vest(_users[15], _users[14], vest_amount, interest_rate_100prcnt));
+    check();
+    BOOST_TEST_MESSAGE("--- " << name{_users[14]}.to_string() << " voted");
+    BOOST_CHECK_EQUAL(success(), addvote(_users[14], {_users[0], "permlink6"}, 10000));
```

Comment:
please add check of the vote object: there should be 1 delegator with 100% interest rate